complaint_search/es_builders.py

```diff
@@ -58,15 +61,28 @@ def build(self):
     ## with_child: set to true if this field has a child field
     ## es_child_field_name: if with_child is true, this holds the field
     ## name of the child used (to be filtered in ES)
-    def _build_bool_should_clauses(self, field):
+    def _build_bool_clauses(self, field):
         es_field_name = self._get_es_name(field)
         value_list = [ 0 if cd.lower() == "no" else 1 for cd in self.params[field] ] \
             if field in self._OPTIONAL_FILTERS_STRING_TO_BOOL else self.params.get(field)
+
         if value_list:
+
+            # MUST filters must be in parallel otherwise they will not execute as intended
+            if field in self._OPTIONAL_FILTERS_MUST:
+                term_list_container = []
+                term_list = [ term_list_container.append({"terms": {es_field_name: [value] }})
+                    for value in value_list]
+
+                return term_list_container
+
+            # The most common property for data is to not have a child element
             if not self._has_child(field):
-                term_list = [ {"terms": {es_field_name: [value]}}
+                term_list_container = {"terms": {es_field_name: [] }}
+                term_list = [ term_list_container["terms"][es_field_name].append(value)
                     for value in value_list ]
-                return {"bool": {"should": term_list}}
```

Comment:
You probably don't need list comprehension here, you are basically trying to get the entire `value_list` as the value for the key `es_field_name`, so you can simplify L81 to 83 by doing so:
`term_list = [ {"terms": {es_field_name: value_list}]`
and change L85 to `return term_list`

Reply:
Because I need to return a `dict` here from the clause, the list comprehension you suggest was not working quite as expected. I switched to a `for / in` as the list comprehension was not necessary (as you indicate).  Let me know if you think that will be OK if perhaps there's a better way to put the inner values of the array comprehension into the `dict` as expected by the query. Thanks!